ios/AppDelegate.swift

```diff
@@ -158,16 +158,24 @@ extension AppDelegate {
             let rpcmodule = RPCModule()
 
             // save the wallet
-            rpcmodule.saveWalletInternal()
-            NSLog("BGTask startBackgroundTask - expirationHandler Save Wallet")
+            do {
+              try rpcmodule.saveWalletInternal()
+              NSLog("BGTask startBackgroundTask - expirationHandler Save Wallet")
+            } catch {
+              NSLog("BGTask startBackgroundTask - expirationHandler Save Wallet error: \(error.localizedDescription)")
+            }
 
             // save the background file
             let timeStamp = Date().timeIntervalSince1970
             let timeStampStr = String(format: "%.0f", timeStamp)
             let jsonBackground = "{\"batches\": \"0\", \"message\": \"Finished OK.\", \"date\": \"\(self.timeStampStrStart ?? "0")\", \"dateEnd\": \"\(timeStampStr)\"}"
-            rpcmodule.saveBackgroundFile(jsonBackground)
-            NSLog("BGTask startBackgroundTask - expirationHandler Save background JSON \(jsonBackground)")
-
+            do {
+              try rpcmodule.saveBackgroundFile(jsonBackground)
+              NSLog("BGTask startBackgroundTask - expirationHandler Save background JSON \(jsonBackground)")
+            } catch {
+              NSLog("BGTask startBackgroundTask - expirationHandler Save background JSON \(jsonBackground) error: \(error.localizedDescription)")
```

Comment:
Also here..  I think we want to rethrow the error/exception that was caught.  Also, we probably want to be as specific as possible about which error we are catching...   especially if we're going to do something other than just log.